src/main/java/com/haxwell/apps/quizki/controllers/QuestionController.java

```diff
@@ -42,4 +48,13 @@ public CreatedQuestionDTO getQuestionById(@PathVariable String id) throws GetQue
 		return qs.getQuestionById(id);
 	}
 	
+	@GetMapping
+	@ResponseBody
+	public ArrayList<CreatedQuestionDTO> getQuestions(
+			@RequestParam(value = "page", defaultValue = "0") int zeroBasedPageNumber,
+			@RequestParam(value = "size", defaultValue = "10") int size) throws GetQuestionException {
+		Pageable pageable = PageRequest.of(zeroBasedPageNumber, size);
+		return qs.getQuestions(pageable);
```

Comment:
So I can understand what is better code why would the Pageable be created here instead of in the service?? I was thinking to make the Service compatible with the parameters the Controller received instead of the parameters the Repo expects. Not sure how to resolve the two.

Reply:
Never mind! :-)

Reply:
The 'why' here is because it makes our test easier to write :) Dealing with Pageable in the service made the test more complicated for no apparent gain. This is a happenstance that I come across often in coding (and testing that code). That being "Should I pass in an object which contains the parameters my service method needs, or should I pass in the parameters?" It depends. In this case, the object makes more sense.